shiftcommerce/reference-site
# Feature/Lazyload Cart when minibag is hovered

### Overview

Related GitHub issue: https://github.com/shiftcommerce/trendy-golf/issues/129

- [x] Reviewed and filled in QA section
- [x] Reviewed and filled in Code Review section
- [x] Reviewed and filled in Dependencies review section
- [x] Documented development process in PR
- [x] Documented approach and thought process in issue

This PR implements cart lazyloading in shift-next and shift-react-components

The cart is now loaded when the minibag is hovered

Also, the cart line item count is set in a cookie, and is updated when the cart is retrieved from the platform

### QA

1. Clear your cookies
2. Refresh page
3. Cart should show as zero
4. Hover over minibag icon, network should show a call to `/getCart`
5. Add something to your cart
6. Refresh page, cart line item count should show as `1`
7. Hover over minibag icon, network should show a call to `/getCart`

Removing items from the cart should also update the cookie

### Code Review

N/A

### Dependencies Review

N/A


## ENV variables to be added

N/A
